tests/client_tests/python/asyncpg/test_asyncpg.py

```diff
@@ -4,45 +4,54 @@
 from crate.qa.tests import NodeProvider
 
 
-async def exec_queries(test, hosts):
-    pool = await asyncpg.create_pool(f'postgres://crate@{hosts}/doc')
-    async with pool.acquire() as conn:
-        result = await conn.execute(
-            "CREATE TABLE t1 (x int primary key, y int)")
-        test.assertEqual(result, 'CREATE 1')
+async def basic_queries(test, conn):
+    """
+    Invoke some basic queries.
+    """
+
+    await conn.execute("DROP TABLE IF EXISTS t1")
+
+    result = await conn.execute(
+        "CREATE TABLE t1 (x int primary key, y int)")
+    test.assertEqual(result, 'CREATE 1')
+
+    result = await conn.execute('INSERT INTO t1 (x) VALUES (?)', 1)
+    test.assertEqual(result, 'INSERT 0 1')
 
-        result = await conn.execute('INSERT INTO t1 (x) VALUES (?)', 1)
-        test.assertEqual(result, 'INSERT 0 1')
+    result = await conn.execute('REFRESH TABLE t1')
+    test.assertEqual(result, 'REFRESH 1')
 
-        result = await conn.execute('REFRESH TABLE t1')
-        test.assertEqual(result, 'REFRESH 1')
+    result = await conn.execute('UPDATE t1 SET y = ?', 2)
+    test.assertEqual(result, 'UPDATE 1')
 
-        result = await conn.execute('UPDATE t1 SET y = ?', 2)
-        test.assertEqual(result, 'UPDATE 1')
+    result = await conn.execute('REFRESH TABLE t1')
+    test.assertEqual(result, 'REFRESH 1')
 
-        result = await conn.execute('REFRESH TABLE t1')
-        test.assertEqual(result, 'REFRESH 1')
+    result = await conn.execute('DELETE FROM t1 WHERE y = ?', 2)
+    test.assertEqual(result, 'DELETE 1')
 
-        result = await conn.execute('DELETE FROM t1 WHERE y = ?', 2)
-        test.assertEqual(result, 'DELETE 1')
+    result = await conn.execute('''
+        INSERT INTO t1 (x) (
+            SELECT col1 FROM unnest([1, 2]) WHERE col1 = ?
+        )
+    ''', 1)
+    test.assertEqual(result, 'INSERT 0 1')
 
-        result = await conn.execute('''
-            INSERT INTO t1 (x) (
-                SELECT col1 FROM unnest([1, 2]) WHERE col1 = ?
-            )
-        ''', 1)
-        test.assertEqual(result, 'INSERT 0 1')
+    result = await conn.execute('''
+        INSERT INTO t1 (x) (
+            SELECT col1 FROM unnest([1, 2]) WHERE col1 = ?)
+        ON CONFLICT (x) DO UPDATE SET y = ?
+    ''', 1, 2)
+    test.assertEqual(result, 'INSERT 0 1')
 
-        result = await conn.execute('''
-            INSERT INTO t1 (x) (
-                SELECT col1 FROM unnest([1, 2]) WHERE col1 = ?)
-            ON CONFLICT (x) DO UPDATE SET y = ?
-        ''', 1, 2)
-        test.assertEqual(result, 'INSERT 0 1')
 
-        result = await conn.fetch('SELECT pg_catalog.pg_get_keywords()')
-        keyword = sorted([row[0] for row in result], key=lambda x: x[0])[0]
-        test.assertEqual(keyword, ('add', 'R', 'reserved'))
+async def binary_streaming(test, conn):
+    """
+    Verify that binary streaming of record types works (2020-02-07).
+    """
```

Comment:
```suggestion
async def test_record_type_can_be_read_using_binary_streaming(test, conn):
```

Reply:
I've implemented this suggestion. However, I did not use the `test_` prefix because those functions are not real test methods yet and I didn't want to risk them to be discovered on their own. Making real test functions out of them would probably need another round of refactoring.